src/main/java/org/reaktivity/nukleus/http/internal/stream/ClientConnectReplyStream.java

```diff
@@ -328,15 +310,22 @@ private void handleDataWhenNotBuffering(
         else
         {
             final OctetsFW payload = data.payload();
-            final int limit = payload.limit();
-            int offset = payload.offset();
-
-            offset = decode(payload.buffer(), offset, limit);
+            if (slotIndex == NO_SLOT)
+            {
+                final int limit = payload.limit();
+                int offset = payload.offset();
+                offset = decode(payload.buffer(), offset, limit);
```

Comment:
Suggest the following:
```
                DirectBuffer buffer = payload.buffer();
                int offset = payload.offset();
                int limit = payload.limit();

                if (slotIndex != NO_SLOT)
                {
                    final MutableDirectBuffer slotBuffer = bufferPool.buffer(slotIndex);
                    slotBuffer.putBytes(slotOffset, buffer, offset, limit - offset);
                    slotOffset += limit - offset;
                    buffer = slotBuffer;
                    offset = 0;
                    limit = slotOffset;
                }

                DecoderState previous = null;
                while (offset <= limit && previous != decodeState)
                {
                    previous = decodeState;
                    offset = decodeState.decode(buffer, offset, limit);
                }

                if (offset < limit)
                {
                    if (slotIndex == NO_SLOT)
                    {
                        slotIndex = factory.bufferPool.acquire(connectReplyId);
                    }

                    if (slotIndex == NO_SLOT)
                    {
                        factory.writer.doReset(connectReplyThrottle, connectRouteId, connectReplyId, factory.supplyTrace.getAsLong());
                        connection.persistent = false;
                        doCleanup(CloseAction.ABORT);
                    }
                    else
                    {
                        final MutableDirectBuffer slotBuffer = bufferPool.buffer(slotIndex);
                        slotBuffer.putBytes(0, buffer, offset, limit - offset);
                        slotOffset = limit - offset;
                    }
                }
                else if (slotIndex != NO_SLOT)
                {
                    factory.bufferPool.release(slotIndex);
                    slotIndex = NO_SLOT;
                    slotOffset = 0;
                }
```
... and remove `slotPosition` as it is no longer needed.

This should be the only method to deal with `slotIndex`, except for cleaning up on `ABORT` or `RESET`.